node/Utils/CheckRLimits.hs

```diff
@@ -0,0 +1,36 @@
+{-# language ScopedTypeVariables #-}
+
+module Utils.CheckRLimits(checkRLimits) where
+
+import Control.Monad
+import Data.Char
+import Data.List(stripPrefix)
+import System.Process
+import Text.Read
+
+checkRLimits :: IO ()
+checkRLimits = do
+    pid <- getCurrentPid
+    limits <- readFile $ "/proc/" <> show pid <> "/limits"
+    let
+        -- the format of /proc/<pid>/limits looks like
+        -- Something      its soft limit     its hard limit    the units of the limit
+        -- for example
+        -- Max open files     1024        1024       files
+        -- we only care about the soft limit, which is the real limit for the process.
+        openFileLimit =
+            [ n
+            | l <- lines limits
+            , Just rest <- [stripPrefix "Max open files" l]
+            , Just (n :: Int) <- [readMaybe (takeWhile isDigit $ dropWhile isSpace rest)]
+            ]
+    case openFileLimit of
+        [n] ->
+            when (n < 32768) $
+                error $
```

Comment:
I wonder if we should just print this message to stdout and exit instead of `error`. Since this is a user facing error from a very specific check, the source information and the indentation from the`error` display is probably unnecessary noise.

Reply:
:+1: